gnomad_qc/v4/resources/basics.py

```diff
@@ -366,3 +367,217 @@ def check_resource_existence(
                     f"rerun {step} with --overwrite. The following files already exist:"
                 ),
             )
+
+
+class PipelineStepResourceCollection:
+    """A collection of resources used in a specific step of a gnomad_qc pipeline."""
+
+    def __init__(
+        self,
+        step_name: str,
+        previous_pipeline_steps: List = [],
```

Comment:
So when I first went through the class, I assumed this would contain any upstream steps , obviously didnt read the doc on L392 since you specify input, but I wondering if theres another name that would make this a bit more clear, like `previous_step_w_input` or something. Again I don't think it's critical but the variable name doesnt mention the steps output is required.